registry/data-models/models.py

```diff
@@ -0,0 +1,168 @@
+      |    +----------------+                      +-----------------+     +----------+
+      |    | DerivedFeature |                      |  AnchorFeature  |          /|\
+      |    +----------------+                      +-----------------+           |
+      |            |                                                             |
+      |            |                                                             |extends
+      |            |                                                       +------------------+
+      |            +-------------------------------------------------------|   FeatureSource  |
+      |                                                                has +------------------+
+      |                                                                             |
+      +-----------------------------------------------------------------------------|
+                      
+                      
+                      
+                      
+                      
+                      
+"""
+
+
+class FeatureId(BaseModel):
+    """
+    Id for Feature, it's unique ID represents Feature.
+    """
+    pass
+
+
+class FeatureNameId(BaseModel):
+    """
+    Id for FeatureName, it's unique ID represents FeatureName.
+    """
+    pass
+
+
+class AnchorId(BaseModel):
+    """
+    Id for Anchor, it's unique ID represents Anchor.
+    """
+    pass
+
+
+class ProjectId(BaseModel):
+    """
+    Id for Project, it's unique ID represents Project.
+    """
+    pass
+
+
+class Source(BaseModel):
+    """
+    Type of FeatureValue.
+    It defines where the feature is extracted from.
+    """
+    pass
+
+
+class DataSource(Source):
+    pass
+
+
+class FeatureSource(Source):
+    input_feature_name_ids: List[FeatureNameId]  # List of input feature name Keys
+    pass
+
+
+class Transformation(BaseModel):
+    """
+    The transformation of a Feature.
+    A transformation function represents the transformation logic to produce feature value from the source of FeatureAnchor
+    """
+    pass
+
+
+class Feature(BaseModel):
+    """
+    Actual implementation of FeatureName.
+    An implementation defines where a feature is extracted from (Source) and how it is computed (Transformation).
+    The Source of a feature can be raw data sources and/or other features.
+    """
+    id: FeatureId  # Unique ID for Feature
+    feature_name_id: FeatureNameId  # Id of the feature name that the feature belongs to
+    source: Source  # Source can be either data source or feature source
+    transformation: Transformation  # transformation logic to produce feature value
+
+
+class AnchorFeature(Feature):
+    """
+    Feature implementation of FeatureName which anchored to a data source.
+    """
+    source: DataSource
+
+
+class DerivedFeature(Feature):
+    """
+    Feature implementation that is derived from other FeatureNames.
+    """
+    source: FeatureSource
+
+
+class FeatureName(BaseModel):
+    """
```

Comment:
nit on documentation

"""
    Named Feature Interface that can be backed by multiple Feature implementations across different environments accessing different sources (data lake access for batch training, KV store access for online serving).
    Each FeatureName is defined by feature producer. Feature consumers reference a feature by that name to access that feature data, agnostic of runtime environment.
    Each FeatureName also encloses attributes that doesn't change across implementations.
"""

Reply:
Thanks, applied the change.